java-checks/src/main/java/org/sonar/java/checks/SubClassStaticReferenceCheck.java

```diff
@@ -99,7 +99,7 @@ public void visitClass(ClassTree tree) {
     @Override
     public void visitIdentifier(IdentifierTree tree) {
       Type type = tree.symbolType();
-      if (!classType.equals(type) && type.isSubtypeOf(classType)) {
+      if (!classType.is(type.fullyQualifiedName()) && type.isSubtypeOf(classType)) {
```

Comment:
Yep, failing test cases on current test file if I don't do the change: https://github.com/SonarSource/sonar-java/blob/master/java-checks/src/test/files/checks/SubClassStaticReferenceCheck.java#L10

With the changes that I've done, using equals can not work anymore, as we will have "classType" compared in case of constructors to the "MethodJavaType" associated with the identifier, which will never be equals, even if the constructor does return the same type. However, the "is" method handle the case properly, as we will use the symbol fully qualified name, which is the correct one.